src/mlpack/methods/kmeans/kmeans_main.cpp

```diff
@@ -242,7 +245,8 @@ void RunKMeans(const InitialPartitionPolicy& ipp)
   // Make sure we have an output file if we're not doing the work in-place.
   RequireAtLeastOnePassed({ "in_place", "output", "centroid" }, false,
       "no results will be saved");
-
+  if (!CLI::HasParam("input"))
+    Log::Fatal << "Must specify input"<< endl;
```

Comment:
`input` is a required parameter, so this shouldn't be needed.

Reply:
Actually when I'm not setting the parameter input then it doesn't throw a Fatal error. Although in CLI it does. I tried to look out for the cause but couldn't find it!

Reply:
This is because `ParseCommandLine()` is what checks for required parameters, and that isn't called for the test bindings since we are calling `mlpackMain()`, so there is no need to test here that a required parameter not being passed causes a fatal error.